apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -300,6 +298,18 @@ export class InstallManager {
         }
       }
 
+      const approvedPackagesChecker: ApprovedPackagesChecker = new ApprovedPackagesChecker(this._rushConfiguration);
+      if (this._options.allowShrinkwrapUpdates) {
```

Comment:
It might just be how my brain works, but to me, this would be less confusing as:
```
if (approvedPackagesChecker.approvedPackagesFilesAreOutOfDate) {
  if (this._options.allowShrinkwrapUpdates) { 
    // ... update & warn
  } else {
    // ... error
  }
}